cuda.spec

```diff
@@ -34,6 +34,7 @@ mkdir -p %{i}/lib64/stubs
 
 # package only the runtime static library
 mv %_builddir/build/lib64/libcudadevrt.a %{i}/lib64/
+mv %_builddir/build/lib64/libcudart_static.a %{i}/lib64/
```

Comment:
Mhm... I don't think that mixing the shared and static version of `libcudart` is a good idea.
Can ONNX not use the dynamic version ?

Reply:
This `libcudart_static.a` is needed otherwise `enable_language(CUDA)` in cmake crashes. I don't think onnxruntime really uses it.

Reply:
why do we need cmake ?

Reply:
can you set the `CUDA_USE_STATIC_CUDA_RUNTIME` cmake option to `OFF` ?

Reply:
> why do we need cmake ?
> it looks like cuDNN is not built, it's simply unpacked.

We use cmake for ONNXRuntime. For cuDNN indeed it's simply unpacked.

> can you set the CUDA_USE_STATIC_CUDA_RUNTIME cmake option to OFF ?

Sure, I can try that.

Reply:
@fwyzard Unfortunately `CUDA_USE_STATIC_CUDA_RUNTIME` does not make a difference. It seems it's superseded by `CMAKE_CUDA_RUNTIME_LIBRARY` now?

In fact I managed to work around this with the following lines:
```
   -DCMAKE_CUDA_FLAGS="-cudart shared" \
   -DCMAKE_CUDA_RUNTIME_LIBRARY=Shared \
   -DCMAKE_TRY_COMPILE_PLATFORM_VARIABLES="CMAKE_CUDA_RUNTIME_LIBRARY" \
```
Now we don't need `libcudart_static.a` anymore. In fact I think the really useful one is `-DCMAKE_CUDA_FLAGS="-cudart shared"`, which is sufficient to solve the problem in a newer cmake version, but somehow I need all three lines in the cmake version we use...

Also the problem is purely in cmake -- when calling `enable_language(CUDA)` it tries to compile a test program with `nvcc` and then parse the output to set up various CUDA paths/flags, and there it links to `libcudart_static.a` since the `-cudart` defaults to `static` in `nvcc`. After that stage, whether linking to cudart statically or dynamically can be controlled by `CMAKE_CUDA_RUNTIME_LIBRARY` and it's set to `Shared` [in onnxruntime](https://github.com/cms-externals/onnxruntime/blob/cms/v1.6.0/cmake/CMakeLists.txt#L1107).